k9mail-library/src/main/java/com/fsck/k9/mail/internet/MessageExtractor.java

```diff
@@ -142,7 +144,10 @@ public static void findViewablesAndAttachments(Part part,
         }
 
         Body body = part.getBody();
-        if (body instanceof Multipart) {
+        if (body instanceof BinaryMemoryBody) {
```

Comment:
the diff here is shown misleading - please check the diff in MessageExtractor.java manually.
i had just added handling of BinaryMemoryBody for the specific case of redirecting.